### DISCUSSION TITLE
Long Query and web search Plugin made broken layout

### DISCUSSION THREAD

**Discussion by onesounds** [Flow-Launcher/Flow.Launcher#736] [General]
![image](https://user-images.githubusercontent.com/6903107/136310481-acd70b69-5aa8-46b5-9b70-4ead44b146bf.png)

- It is a bug that is rare in real use but clearly noticeable.
- It is an old heritage that has existed since Wox.

**To Reproduce**
1. Turn on Web Search Plugin
2. Write Long query till end to window
3. Broken Layout

- Search Plugin Turn off and Write long query, It solved.
- in, these case. It showing same issue with other plugin (like shell) too.

**Hint**
- If you look closely, Suggest (Textbox in beck) does not scroll to the end. 
- It looks broken because the scroll position is not correct. It(suggestquerybox) need sync to querybox.
- https://stackoverflow.com/questions/4055720/scrolling-to-the-end-of-a-single-line-wpf-textbox
- https://www.py4u.net/discuss/1451456
- in this case, suggest textbox has a little different clear type setting. so query box text little blink the  in this case.
-  I tried turn off all clear type setting in style xaml. but It still happen. I never found other code.
- Something in plugin touches the layout.

**Comment by Zeroto521** [comment]
same to #153

**Comment by jjw24** [comment]
i have a feeling it's the suggestion(autocomplete) textbox that's causing this.

What happens if you delete suggestion textbox?

**Reply by onesounds** [reply]
Of course, it comes out well. I think this function is meaningless and designally bad, so I think it's good if it can be erased.

When I tried it on debug, there was an error.
![image](https://user-images.githubusercontent.com/6903107/142074587-ff8a31b9-22d4-4745-9c0f-a18305fe8d3d.png)

**Reply by jjw24** [reply]
hm other launchers have auto-complete. I think we need to rework this, or do we just want to remove this autocomplete altogether?

**Reply by onesounds** [reply]
powertoy only maintained this feature. and little slightly different. they use searchbox to display file name when choosing item.

This applies only to files or paths, and in this case, the auto-completed items will already appear at the beginning of the list. In other words, being recommended for automatic completion means that it has already been selected as a top priority in the search results. The action I need is already possible here. I don't know why it exists because I don't use that function. example?

**Reply by jjw24** [reply]
yeah i think tab to autocomplete the result's title/subtitle/some other property into search window is sufficient. I personally dont find it that useful either aside from it being visual.

**Comment by Garulf** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/pull/854

While not a "Fix" may help with this issue.
